### PR TITLE
Update typescript to 4.9 and @types/node to latest 16.x

### LINKED ISSUE #12151: Upgrade typescript
Currently there are type-related errors in telemetryGenerator.node.ts by default in VS Code as it uses the bundled version of tsserver (4.9), not the workspace version (4.7). We should upgrade typescript and fix the compile errors.

### PR DESCRIPTION
**Pull request by DonJayamanne**
Fixes #12151

### CODE REVIEW

File: build/ci/postInstall.js [hunk 1]
```diff
@@ -145,6 +145,23 @@ exports.javascript = {
...
+function fixUIFabricForTS49() {
+    [
+        'node_modules/@uifabric/merge-styles/lib/mergeStyleSets.d.ts',
+        'node_modules/@uifabric/utilities/lib/styled.d.ts'
+    ].forEach((file) => {
+        const filePath = path.join(__dirname, '..', '..', file);
+        if (!fs.existsSync(filePath)) {
+            return;
+        }
+        const contents = fs.readFileSync(filePath, 'utf8').toString();
```
Comment (DonJayamanne) [inline build/ci/postInstall.js:157]: This package is not ready for ts 4.9 yet.

File: package.json [hunk 2]
```diff
@@ -2216,7 +2216,7 @@
         "@types/memoize-one": "^4.1.1",
         "@types/mocha": "^9.1.0",
         "@types/nock": "^10.0.3",
-        "@types/node": "^14.17.4",
+        "@types/node": "^16.18.6",
```
Comment (DonJayamanne) [inline package.json:2219]: Updated @types/node

File: src/kernels/jupyter/jupyterCellOutputMimeTypeTracker.ts [hunk 3]
```diff
@@ -3,14 +3,13 @@
 
 'use strict';
 
-import { IDisposable } from '@fluentui/react';
```
Comment (DonJayamanne) [inline src/kernels/jupyter/jupyterCellOutputMimeTypeTracker.ts:6]: Oops

File: src/kernels/raw/launcher/kernelProcess.node.ts [hunk 4]
```diff
@@ -131,7 +131,9 @@ export class KernelProcess implements IKernelProcess {
             !this.interrupter
         ) {
             traceInfo('Interrupting kernel via SIGINT');
-            kill(this._process.pid, 'SIGINT');
+            if (this._process.pid) {
```
Comment (DonJayamanne) [inline src/kernels/raw/launcher/kernelProcess.node.ts:134]: As a result of using the latest @types/node@16.x

File: src/platform/common/process/logger.node.ts [hunk 5]
```diff
@@ -37,7 +37,7 @@ export class ProcessLogger implements IProcessLogger {
 
         const info = [`> ${removeHomeFromFile(file)} ${argsList}`];
         if (options && options.cwd) {
-            info.push(`${Logging.currentWorkingDirectory()} ${removeHomeFromFile(options.cwd)}`);
+            info.push(`${Logging.currentWorkingDirectory()} ${removeHomeFromFile(options.cwd.toString())}`);
```
Comment (DonJayamanne) [inline src/platform/common/process/logger.node.ts:40]: As a result of using the latest @types/node@16.x

File: src/telemetryGenerator.node.ts [hunk 6]
```diff
@@ -208,12 +208,13 @@ function computePropertiesForLiteralType(literalType: ts.TypeLiteralNode, typeCh
             if (gdprEntryOfCurrentlyComputingTelemetryEventName) {
                 if (
                     'properties' in gdprEntryOfCurrentlyComputingTelemetryEventName[1] &&
-                    name in gdprEntryOfCurrentlyComputingTelemetryEventName[1]['properties']
+                    gdprEntryOfCurrentlyComputingTelemetryEventName[1]['properties'] &&
+                    name in (gdprEntryOfCurrentlyComputingTelemetryEventName[1]['properties'] as any)
```
Comment (DonJayamanne) [inline src/telemetryGenerator.node.ts:212]: TS 4.9 compilation issues

File: types/slickgrid/index.d.ts [hunk 7]
```diff
@@ -3,6 +3,7 @@
 // Definitions by: Josh Baldwin <https://github.com/jbaldwin>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.3
+// @ts-nocheck
```
Comment (DonJayamanne) [inline types/slickgrid/index.d.ts:6]: This d.ts file isn't ready for ts 4.9 (hopefully this file will be removed soon from the repo)

**Review by mjbvz — APPROVED** [review]

**Pull request merged**